rust/src/session.rs

```diff
@@ -566,8 +566,9 @@ impl SessionManager {
             )
             .await?;
             session.start(settings, telemetry_command_tx).await?;
-            session_map.insert(endpoint_url.clone(), session.clone());
-            Ok(session)
+            let cloned_session = session.clone();
```

Comment:
It looks like the clone differs from the original, which is somewhat weired and should be avoided, see https://doc.rust-lang.org/stable/std/clone/trait.Clone.html#how-can-i-implement-clone

We need to modify `Session` to make itself and its clones are idential.